fineract-provider/src/main/java/org/apache/fineract/adhocquery/domain/AdHoc.java

```diff
@@ -0,0 +1,156 @@
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.adhocquery.domain;
+
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+
+import org.apache.commons.lang.BooleanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.fineract.adhocquery.api.AdHocJsonInputParams;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.domain.AbstractAuditableCustom;
+import org.apache.fineract.infrastructure.core.domain.JdbcSupport;
+import org.apache.fineract.useradministration.domain.AppUser;
+import org.joda.time.LocalDate;
+
+
+@Entity
+@Table(name = "m_adhoc")
+public class AdHoc extends AbstractAuditableCustom<AppUser, Long> {
+
+  	@Column(name = "name", length = 100)
+    private  String name;
+    
+    @Column(name = "query", length = 2000)
+    private  String query;
+   	
+    @Column(name = "table_name", length = 100)
+    private  String tableName;
+    
+    
+    @Column(name = "table_fields", length = 2000)
+    private  String tableFields;
+   	
+    @Column(name = "email", length = 500)
+    private  String email;
+    
+   
+	@Column(name = "IsActive", nullable = false)
+    private boolean isActive = false;
+   	
+    private AdHoc(final String name, final String query,final String tableName,final String tableFields ,final String email,final boolean isActive) {
+        this.name = StringUtils.defaultIfEmpty(name, null);
+        this.query=StringUtils.defaultIfEmpty(query,null);
+        this.tableName=StringUtils.defaultIfEmpty(tableName,null);
+        this.tableFields=StringUtils.defaultIfEmpty(tableFields,null);
+        this.email=StringUtils.defaultIfEmpty(email,null);
+        this.isActive = BooleanUtils.toBooleanDefaultIfNull(isActive, false);
+       
+    }
+    public static AdHoc fromJson(final JsonCommand command) {
+        final String name = command.stringValueOfParameterNamed(AdHocJsonInputParams.NAME.getValue());
+        final String query = command.stringValueOfParameterNamed(AdHocJsonInputParams.QUERY.getValue());
```

Comment:
String commandQuery=command.stringValueOfParameterNamed(AdHocJsonInputParams.QUERY.getValue());
        if(commandQuery.toLowerCase().contains("drop")||commandQuery.toLowerCase().contains("delete")){
        	commandQuery=null;
        }
      

do you thinks this will be ok